sklearn/metrics/tests/test_common.py

```diff
@@ -1743,9 +1743,14 @@ def check_array_api_metric(
 
     with config_context(array_api_dispatch=True):
         if sample_weight is not None:
-            sample_weight = xp.asarray(sample_weight, device=device)
+            sample_weight = xp.asarray(sample_weight.astype(dtype), device=device)
         metric_xp = metric(y_true_xp, y_pred_xp, sample_weight=sample_weight)
 
+        if not isinstance(metric_xp, float):
+            # If the result is not a scalar, the array has to be in the CPU
+            # before transforming it to a numpy array.
+            metric_xp = xp.asarray(metric_xp, device="cpu")
```

Comment:
The "cpu" device is not always valid. We should instead use the `_convert_to_numpy` helper in such tests.